src/platform-implementation-js/dom-driver/gmail/views/gmail-thread-row-view.ts

```diff
@@ -430,28 +437,28 @@ class GmailThreadRowView {
       });
   }
 
-  addImage(
-    inIconDescriptor:
-      | ImageDescriptor
-      | Kefir.Observable<ImageDescriptor | null, any>,
-  ) {
+  addImage(inIconDescriptor: Descriptor<ImageDescriptor | null>) {
     if (!this._elements.length) {
       console.warn('addImage called on destroyed thread row');
       return;
     }
 
-    const prop = kefirCast(Kefir, inIconDescriptor)
+    const prop: Kefir.Observable<[ImageDescriptor | null], unknown> = kefirCast(
+      Kefir,
+      inIconDescriptor,
+    )
```

Comment:
to avoid any casting, we can cast only the kefirCast result

```
kefirCast(Kefir, inIconDescriptor) as Kefir.Observable<ImageDescriptor | null, unknown>
```

and then add a second paramter to `combine` to define the combinator explicitly
```
.combine(
    Kefir.merge(....),
    (iconDescriptor, _) => [iconDescriptor],
);
```